lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString.apply(this, arguments);
```

Comment:
Wouldn't it be faster to do `return slowToString.call(this, arguments[0], arguments[1], arguments[2]);`? Or maybe pass `this` as the first argument and avoid `.call()`/`.apply()` altogether?